pubnub-core/src/data/channel.rs

```diff
@@ -0,0 +1,89 @@
+//! Channel related types.
+
+use std::convert::TryFrom;
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+const NAME_PROHIBITED_SYMBOLS: &[char] = &[',', '/', '\\', '.', '*', ':'];
```

Comment:
Actually, I managed to find the documentation on restricted characters in channels: https://www.pubnub.com/docs/web-javascript/data-streams-publish-and-subscribe#channel_name_design_consideration

Reply:
:joy: "By convention only."

Reply:
Yep, I've already corrected the actual implementation as we discussed, as added a huge explainer at the doc comment. :smile:

Reply:
Even if the Rust SDK follows the convention, that does not mean that every client will. We should be very very careful about how we deal with errors. Channel names are controlled by remote clients, not by the PubNub server, and not by the Rust SDK.

The most obvious examples are when subscribing to wildcard channels, and channel groups. Both of these add a layer of indirection that will allow an SDK written "to spec" to be given invalid channel names by remote clients.

Reply:
I have this addressed in the upcoming PR. Due to effectively technical difficulties with access, I can't push further changes to this PR, unfortunately.
I propose we just merge this one, then address the pending issues from here in the next, carrying on the change requests.
For the same reasons, I'll need your help with actual merging, please do a rebase merge so that I can apply my changes atop of master.
We should talk about the process for the next few weeks during our weekly sync up.

Reply:
Created #39, so that we can validate it fixes the issues from this PR.

Reply:
Rebase-merging now...